product_configurator_wizard/wizard/product_configurator.py

```diff
@@ -121,19 +138,31 @@ def get_form_vals(self, dynamic_fields, domains):
         """
         vals = {}
 
-        dynamic_fields = {k: v for k, v in dynamic_fields.iteritems() if v}
+        dynamic_fields = dynamic_fields.copy()
 
         for k, v in dynamic_fields.iteritems():
-            if not v:
-                continue
+            attribute_id = int(k.replace(self.field_prefix, ''))
+            tmpl_attribute = self.product_tmpl_id.attribute_line_ids.filtered(
+                lambda a: a.attribute_id.id == attribute_id)
             available_val_ids = domains[k][0][2]
-            if isinstance(v, list):
+
+            vals.update(
+                self.get_support_fields(k, available_val_ids, tmpl_attribute)
+            )
+
+            if v and isinstance(v, list):
                 value_ids = list(set(v[0][2]) & set(available_val_ids))
                 dynamic_fields.update({k: value_ids})
                 vals[k] = [[6, 0, value_ids]]
-            elif v not in available_val_ids:
+            elif v and v not in available_val_ids:
                 dynamic_fields.update({k: None})
                 vals[k] = None
+                v = None
+            if not v:
+                if tmpl_attribute.required and not tmpl_attribute.multi and \
```

Comment:
In this case, I am allowing for the fact that an attribute may have been invalid and cleared.  If, after this, there is only one value available, and it is a required attribute, then auto select that only value.